src/API/Endpoints/Reports/Endpoint.php

```diff
@@ -361,10 +361,13 @@ public function getPayments( $startStr, $endStr, $orderBy = 'date', $number = -
 		$gateway = array_keys( $gatewayObjects );
 
 		$args = [
-			'post_status' => [
-				'publish',
-				'give_subscription',
-			],
+			'post_status' => apply_filters(
+				'give_reports_add_post_status',
+				[
+					'publish',
+					'give_subscription',
+				]
+			),
```

Comment:
@mehul0810 Thanks for the PR! Instead of just filtering the status, let's apply a `give_reports_get_payments_query_args` filter to the whole `$args` array.

Reply:
@JasonTheAdams Just wondering whether as this is admin reports will filtering the whole `$args` array would be useful? 🤔  I thought about it and it seems to me that if I filter the whole `$args` incorrectly then it can break the reports completely.

Let me know what you think.

Reply:
@JasonTheAdams I think the suggestion you provided is good to implement as filtering `$args` will help specially if anyone doing major customizations with GiveWP. So, I am doing the change just now.

Reply:
If people are going to break things, this filter won't stop them. 😉

Reply:
@JasonTheAdams Additionally, I noticed that just adding the filter is not working out as condition for `publish` and `give_subscription` for reports is scattered across smaller modules present within the reports page as in the screenshot. I believe I can achieve it using a helper fn which will implement the conditional logic and the function used instead of the conditional check everywhere but this can take time to implement and test the functionality. So, I will revert back by tomorrow.

![image](https://user-images.githubusercontent.com/1852711/95394163-84b1a880-0919-11eb-97b4-ef67f5834dee.png)